pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -693,6 +694,38 @@ func (r *MachineSetSyncReconciler) createOrUpdateCAPIMachineSet(ctx context.Cont
 	return ctrl.Result{}, nil
 }
 
+// ensureSyncFinalizer ensures the sync finalizer is present across mapi and capi machine sets.
+// It attempts to set both in one call, aggregating errors.
+func (r *MachineSetSyncReconciler) ensureSyncFinalizer(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet) (bool, error) {
+	var shouldRequeue bool
+
+	var errors []error
+
+	if mapiMachineSet != nil {
+		if mapiMachineSet.DeletionTimestamp.IsZero() {
```

Comment:
I haven't paid close attention to this work, but it looks like we're assuming that the deletion timestamp will always be set for both.

Is this field synced across, so that when the authoritative side is marked, the mirror is too?